analyzers/tests/SonarAnalyzer.UnitTest/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.cs

```diff
@@ -928,19 +974,21 @@ private class Context
 
             public Context(string code, AnalyzerLanguage language, string localFunctionName = null)
             {
-                IMethodSymbol originalDeclaration;
                 Cfg = TestHelper.CompileCfg(code, language, code.Contains("// Error CS"));
-                if (localFunctionName == null)
+                if (localFunctionName != null)
                 {
-                    originalDeclaration = (IMethodSymbol)Cfg.OriginalOperation.SemanticModel.GetDeclaredSymbol(Cfg.OriginalOperation.Syntax);
-                }
-                else
-                {
-                    originalDeclaration = Cfg.LocalFunctions.Single(x => x.Name == localFunctionName);
-                    Cfg = Cfg.GetLocalFunctionControlFlowGraph(originalDeclaration);
+                    Cfg = Cfg.GetLocalFunctionControlFlowGraph(Cfg.LocalFunctions.Single(x => x.Name == localFunctionName));
                 }
                 Console.WriteLine(CfgSerializer.Serialize(Cfg));
-                Lva = new RoslynLiveVariableAnalysis(Cfg, originalDeclaration);
+                Lva = new RoslynLiveVariableAnalysis(Cfg);
+            }
+
+            public Context(string code, SyntaxKind syntaxKind)
+            {
+                var (tree, model) = TestHelper.Compile(code, false, AnalyzerLanguage.CSharp);
+                var method = tree.GetRoot().DescendantNodes().First(x => x.RawKind == (int)syntaxKind);
```

Comment:
It's not always a method.
```suggestion
                var node = tree.GetRoot().DescendantNodes().First(x => x.RawKind == (int)syntaxKind);
```